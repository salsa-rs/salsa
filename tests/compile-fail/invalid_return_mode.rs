use salsa::Database as Db;

#[salsa::input]
struct MyInput {
    #[returns(clone)]
    text: String,
}

#[salsa::tracked(returns(not_a_return_mode))]
fn tracked_fn_invalid_return_mode(db: &dyn Db, input: MyInput) -> String {
    input.text(db)
}

#[salsa::input]
struct MyInvalidInput {
    #[returns(not_a_return_mode)]
    text: String,
}

fn main() { }