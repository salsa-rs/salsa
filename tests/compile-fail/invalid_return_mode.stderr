error: Invalid return mode. Allowed modes are: ["copy", "clone", "ref", "deref", "as_ref", "as_deref"]
 --> tests/compile-fail/invalid_return_mode.rs:9:26
  |
9 | #[salsa::tracked(returns(not_a_return_mode))]
  |                          ^^^^^^^^^^^^^^^^^

error: Invalid return mode. Allowed modes are: ["copy", "clone", "ref", "deref", "as_ref", "as_deref"]
  --> tests/compile-fail/invalid_return_mode.rs:16:15
   |
16 |     #[returns(not_a_return_mode)]
   |               ^^^^^^^^^^^^^^^^^

error: cannot find attribute `returns` in this scope
  --> tests/compile-fail/invalid_return_mode.rs:16:7
   |
16 |     #[returns(not_a_return_mode)]
   |       ^^^^^^^
